eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -1249,6 +1249,54 @@ public function testLoadLocationChildrenDataWithOffsetAndLimit(LocationList $loc
         );
     }
 
+    /**
+     * @covers \eZ\Publish\API\Repository\LocationService::loadLocationChildren
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function testLoadLocationChildrenRespectsParentSortingClauses(): void
+    {
+        $repository = $this->getRepository();
+        $locationService = $repository->getLocationService();
+
+        $parentLocation = $locationService->loadLocation(5);
+
+        // Update Location in order to change sort clauses to Name DESC
+        $locationUpdateStruct = $locationService->newLocationUpdateStruct();
+        $locationUpdateStruct->sortField = Location::SORT_FIELD_NAME;
+        $locationUpdateStruct->sortOrder = Location::SORT_ORDER_DESC;
+        $parentLocation = $locationService->updateLocation(
+            $parentLocation,
+            $locationUpdateStruct
+        );
+
+        $children = $locationService->loadLocationChildren($parentLocation);
+        $namesWithDefaultSorting = [];
+        foreach ($children as $child) {
+            $namesWithDefaultSorting[] = $child->getContentInfo()->name;
+        }
+
+        // Update Location in order to change sort clauses to NAME ASC
+        $locationUpdateStruct = $locationService->newLocationUpdateStruct();
+        $locationUpdateStruct->sortField = Location::SORT_FIELD_NAME;
+        $locationUpdateStruct->sortOrder = Location::SORT_ORDER_ASC;
+        $parentLocation = $locationService->updateLocation(
+            $parentLocation,
+            $locationUpdateStruct
+        );
+
+        $children = $locationService->loadLocationChildren($parentLocation);
+
+        $namesWithContentSorting = [];
+        foreach ($children as $child) {
+            $namesWithContentSorting[] = $child->getContentInfo()->name;
+        }
+
+        self::assertNotEquals($namesWithDefaultSorting, $namesWithContentSorting);
```

Comment:
Can we fix the ordering and compare to a snapshot? Just checking that they are not equals with default ordering seems... not really accurate.

Reply:
There is no default ordering here because I'm specifically setting sort clauses twice.

Reply:
>> Can we fix the ordering and compare to a snapshot? Just checking that they are not equals with default ordering seems... not really accurate.

> There is no default ordering here because I'm specifically setting sort clauses twice.

What we're rather aiming here is that we:
1. Create sub items with specific items, taken from an array
2. Give proper sorting clauses
3. Check if the loaded list of names is the same as expected list in the context of sort clause

Sth like:
```php
        $childrenNames = array_map(
            static function (Location $location) {
                return $location->getContentInfo()->name;
            },
            iterator_to_array($locationService->loadLocationChildren($parentLocation))
        );
       self::assertSame($expectedChildrenNames, $childrenNames)
```
where `$expectedChildrenNames` depend on a chosen and set sorting field and order.

Reply:
Done in https://github.com/ezsystems/ezplatform-kernel/pull/351/commits/4d1917dd4115b9e74548ec5cfcd6f6207ace2c7b